src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java

```diff
@@ -96,4 +101,32 @@ private void checkReference(HashId reference, UnIterableMap<HashId, Integer> rat
             throw new InvalidAlgorithmParameterException(REFERENCE_TRANSACTION_TOO_OLD);
         }
     }
+
+    @Override
+    public List<Double> getConfidences(List<Hash> transactions) throws Exception {
+        Hash entryPoint = entryPointSelector.getEntryPoint();
+        UnIterableMap<HashId, Integer> rating = ratingCalculator.calculate(entryPoint);
+
+        List<Hash> tips = new ArrayList<Hash>();
+        for (int i = 0; i < numberOfTipsInGetConfidences; i++) {
```

Comment:
this section, w/ some boilerplate is present in the regular tip selection, why not extract that logic into a method instead of duplicating the code/logic?

Reply:
oh, because `WalkValidatorImpl` is reset in each call.

Reply:
I have a question.
Why numberOfTipsInGetConfidences is 20?
In white paper , it is 100.

Reply:
@KevinKu It's an arbitrary number, no particular reason. It just affects the resolution of the response.